pkgs/applications/audio/squeezelite/default.nix

```diff
@@ -1,41 +1,51 @@
-{ lib, stdenv, fetchFromGitHub
-, alsaLib, flac, libmad, libvorbis, mpg123
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsaLib
+, flac
+, libmad
+, libpulseaudio
+, libvorbis
+, mpg123
+, audioBackend ? "alsa"
 , dsdSupport ? true
-, faad2Support ? true, faad2
-, ffmpegSupport ? true, ffmpeg_3
-, opusSupport ? true, opusfile
-, resampleSupport ? true, soxr
-, sslSupport ? true, openssl
+, faad2Support ? true
+, faad2
+, ffmpegSupport ? true
+, ffmpeg
+, opusSupport ? true
+, opusfile
+, resampleSupport ? true
+, soxr
+, sslSupport ? true
+, openssl
 }:
-
 let
-  concatStringsSep = lib.concatStringsSep;
-  optional = lib.optional;
-  opts = [ "-DLINKALL" ]
-    ++ optional dsdSupport "-DDSD"
-    ++ optional (!faad2Support) "-DNO_FAAD"
-    ++ optional ffmpegSupport "-DFFMPEG"
-    ++ optional opusSupport "-DOPUS"
-    ++ optional resampleSupport "-DRESAMPLE"
-    ++ optional sslSupport "-DUSE_SSL";
+  inherit (lib) optional optionalString;
+
+  pulseSupport = audioBackend == "pulse";
 
-in stdenv.mkDerivation {
-  pname = "squeezelite";
+  name = "squeezelite${optionalString pulseSupport "-pulse"}";
 
+in
+stdenv.mkDerivation {
+  # the nixos module expects the pname to be identical to the binary name
+  pname = name;
```

Comment:
I am somewhat sure that this will break updates for nix-env.

Reply:
Upgrades with `nix-env` should be fine as the name doesn't change (we only add a new one which wasn't there before in case of pulseaudio support).